matchmaking.py

```diff
@@ -146,6 +146,19 @@ def update_user_profile(self) -> None:
             except Exception:
                 pass
 
+    def get_weights(self, online_bots: list[USER_PROFILE_TYPE], rating_preference: str, min_rating: int, max_rating: int,
+                    game_type: str) -> list[int]:
+        """Get the weight for each bot. A higher weights means the bot is more likely to get challenged."""
+        weights = [1] * len(online_bots)

```

Comment:
This is mostly about style, but I don't quite like declaring the default case upfront, where it is not clear yet that it's only a fallback. To me it read as if it was always used, but it's only used if neither "high" nor "low" apply. So why not like this:

```python
if rating_preference == "high":
    return [... for bot in online_bots]
elif rating_preference == "low":
    return [... for bot in online_bots]
else:
    return [1] * len(online_bots)
```

To me, this makes it clearer that there are three separate cases to consider.